satpy/readers/yaml_reader.py

```diff
@@ -1317,6 +1338,129 @@ def _get_new_areadef_heights(self, previous_area, previous_seg_size, **kwargs):
 
         return new_height_proj_coord, new_height_px
 
+    def _new_filehandler_instances(self, filetype_info, filename_items, fh_kwargs=None):
```

Comment:
I started that way at some point, but was not sure how to proceed.  Suppose we have `GEOPreloadableSegmentYAMLReader` (GPSYR) as a subclass of `GEOSegmentYAMLReader` (GSYR), then some questions arise:

- Would we change the five readers that currently use GSYR directly to use GPSYR instead, or only those where we know pre-loading works?  Probably only those where it works.
- How do we combie GPSYR with `GEOVariableSegmentYAMLReader` (GVSYR)?  Does GVSYR subclass GPSYR, suggesting that every GVSYR support pre-loading?  Currently, this is true (`fci_l1c_nc` is the only GVSYR), but perhaps not in the future.  Or are GVSYR and GPSYR bot subclasses of GSYR, and then a dedicated `GEOPreloadableVariableSegmentYAMLReader` (GVSYR) would subclass both GVSYR and GPSYR, leading to diamond rule inheritance?  I prefer to avoid the diamond problem.  Or is the preloadable yaml reader a mixin that doesn't subclass GSYR at all?

I couldn't decide and concluded that I can run away and avoid those questions by now creating a class at all 😄